tools/circleci/requirements.txt

```diff
@@ -0,0 +1,3 @@
+pip>=18.0, <19.0
+setuptools>=38.0, <39.0
```

Comment:
These requirements are quite old, can we upgrade them to newer versions?

Reply:
(sort of) done in a2f0ff9:
* `pip>=20.0` and `setuptools>=38.0` to be consistent with other places we do this. Since setuptools is apparently now at 59.6 (!) this is still pretty old as a lower bound, but I'll leave it to another PR to update that everywhere
* `twine~=3.0` - currently twine is at 3.7.1
```